klib/Makefile

```diff
@@ -155,6 +155,6 @@ CFLAGS+=	$(KERNCFLAGS) -O3 $(INCLUDES) -fPIC $(DEFINES)
 
 # TODO should add stack protection to klibs...
 CFLAGS+=	-fno-stack-protector
-LDFLAGS+=	-pie -nostdlib -T$(ARCHDIR)/klib.lds
+LDFLAGS+=	-shared -Bsymbolic -nostdlib -T$(ARCHDIR)/klib.lds
```

Comment:
It looks like -Bsymbolic flag forces resolution of global symbols to try to come from the shared library. Is this necessary because there are some global name conflicts between some of the klibs and the kernel?

Reply:
No, I didn't find any global name conflicts between the klibs and the kernel. -Bsymbolic is not strictly necessary, even if there are name conflicts, but removes the need to apply relocations to symbols defined within a klib. Without this flag, when loading a klib, for each symbol defined in the klib we have to put the relocated symbol address in the PLT (Procedure Linkage Table), and there is also a runtime performance penalty because each call to global functions must go through the PLT (much like using a table of function pointers); with -Bsymbolic, all calls to functions defined in a klib are direct calls using position-relative addressing, and the PLT is only used for calls to kernel functions.